app/src/db/utils/yars.ts

```diff
@@ -0,0 +1,482 @@
+      .first('role_id');
+  }
+
+  if (actions.includes(Action.READ)) {
+    viewerRole = await knex('yars.role')
+      .where({ name: `${resource}_VIEWER` })
+      .first('role_id');
+  }
+
+  if (actions.includes(Action.UPDATE) || actions.includes(Action.DELETE)) {
+    editorRole = await knex('yars.role')
+      .where({ name: `${resource}_EDITOR` })
+      .first('role_id');
+  }
+
+  if (creatorRole) {
+    await knex('yars.group_role').where({ role_id: creatorRole.role_id }).del();
+  }
+  if (viewerRole) {
+    await knex('yars.group_role').where({ role_id: viewerRole.role_id }).del();
+  }
+  if (editorRole) {
+    await knex('yars.group_role').where({ role_id: editorRole.role_id }).del();
+  }
+}
+
+/**
+ * Deletes role_policy rows
+ * @param knex Knex instance
+ * @param resource Name of the resource
+ * @param actions Array of actions to delete
+ * @returns A promise that resolves when the database calls are complete
+ */
+export async function deleteRolePolicies(knex: Knex, resource: Resource, actions: Action[]) {
+  let creatorRole, viewerRole, editorRole;
+
+  if (actions.includes(Action.CREATE)) {
+    creatorRole = await knex('yars.role')
+      .where({ name: `${resource}_CREATOR` })
+      .first('role_id');
+  }
+
+  if (actions.includes(Action.READ)) {
+    viewerRole = await knex('yars.role')
+      .where({ name: `${resource}_VIEWER` })
+      .first('role_id');
+  }
+
+  if (actions.includes(Action.UPDATE) || actions.includes(Action.DELETE)) {
+    editorRole = await knex('yars.role')
+      .where({ name: `${resource}_EDITOR` })
+      .first('role_id');
+  }
+
+  if (creatorRole) {
+    await knex('yars.role_policy').where({ role_id: creatorRole.role_id }).del();
+  }
+  if (viewerRole) {
+    await knex('yars.role_policy').where({ role_id: viewerRole.role_id }).del();
+  }
+  if (editorRole) {
+    await knex('yars.role_policy').where({ role_id: editorRole.role_id }).del();
+  }
+}
+
+/**
+ * Deletes role rows
+ * @param knex Knex instance
+ * @param resource Name of the resource
+ * @param actions Array of actions to delete
+ * @returns A promise that resolves when the database calls are complete
+ */
+export async function deleteRoles(knex: Knex, resource: Resource, actions: Action[]) {
+  if (actions.includes(Action.CREATE)) {
+    await knex('yars.role')
+      .where({ name: `${resource}_CREATOR` })
+      .del();
+  }
+
+  if (actions.includes(Action.READ)) {
+    await knex('yars.role')
+      .where({ name: `${resource}_VIEWER` })
+      .del();
+  }
+
+  if (actions.includes(Action.UPDATE) || actions.includes(Action.DELETE)) {
+    await knex('yars.role')
+      .where({ name: `${resource}_EDITOR` })
+      .del();
+  }
+}
+
+/**
+ * Deletes policy rows
+ * @param knex Knex instance
+ * @param resource Array of resource names to delete
+ * @returns The generated Knex function to delete data
+ */
+export async function deletePolicies(knex: Knex, resource: Resource[]) {
+  Promise.all(
```

Comment:
Should return promise all here. Also change function param `resource` -> `resources`